### PR TITLE
MM-50202 Removing remaining usage of findDOMNode

### PR DESCRIPTION
**Pull request by hmhealey**
Finishing up the last bit of an old campaign we ran. I had to move the refs in a couple places, but those all appear to work as before.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50202

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by M-ZubairAhmed — APPROVED** [review]

File: components/help/help_controller.tsx [hunk 1]
```diff
@@ -18,47 +17,46 @@ type Props = {
...
-        const helpControllerNode = ReactDOM.findDOMNode(this);
+export default function HelpController(props: Props) {
+    const containerRef = useRef<HTMLDivElement>(null);
 
-        if (helpControllerNode && helpControllerNode instanceof HTMLDivElement) {
-            helpControllerNode.scrollIntoView();
-        }
-    }
+    useEffect(() => {
+        containerRef.current?.scrollIntoView();
```
Comment (michelengelen) [inline components/help/help_controller.tsx:24]: nit: We do have a package to support smooth scrolling and `scrollIntoViewIfNeeded` that is being used in the auto-height-switcher component. If this is something we want to implement here as well. :D

**Review by michelengelen — APPROVED** [review]
lgtm

**Review by koox00 — APPROVED** [review]

**Comment by hmhealey** [comment]
Assigning this to myself while we figure out the new QA situation

File: components/help/help_controller.tsx (hunk 1 above)
Comment (hmhealey) [reply components/help/help_controller.tsx:24]: We don't actually want smooth scrolling here. I forget the exact reason for this, but I believe it was something related to scrolling to the top when this page loads. Also, these pages are actually inaccessible in the app, and I'm going to file a ticket to remove them

**Comment by hmhealey** [comment]
I forgot why I hadn't merged this yet, and it's because this actually breaks the SuggestionList. The code that scrolls to the selected item is currently broken because the ref that it has isn't the DOM node which it needs to scroll

**Pull request closed without merging**